contracts/oracle/CollateralizationMemoizer.sol

```diff
@@ -0,0 +1,204 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./IOracle.sol";
+import "./ICollateralizationOracle.sol";
+import "../refs/CoreRef.sol";
+
+interface IPausable {
+    function paused() external view returns (bool);
+}
+
+/// @title Fei Protocol's Collateralization Oracle
+/// @author eswak
+/// @notice Reads a list of PCVDeposit that report their amount of collateral
+///         and the amount of protocol-owned FEI they manage, to deduce the
+///         protocol-wide collateralization ratio.
+contract CollateralizationMemoizer is IOracle, CoreRef {
+    using Decimal for Decimal.D256;
+
+    // ----------- Events -----------
+
+    event CachedValueUpdate(
+        address from,
+        uint256 indexed protocolControlledValue,
+        uint256 indexed userCirculatingFei,
+        int256 indexed protocolEquity
+    );
+
+    // ----------- Properties -----------
+
+    /// @notice address of the CollateralizationOracle to memoize
+    address public collateralizationOracle;
+
+    /// @notice cached value of the Protocol Controlled Value
+    uint256 public cachedProtocolControlledValue;
+    /// @notice cached value of the User Circulating FEI
+    uint256 public cachedUserCirculatingFei;
+    /// @notice cached value of the Protocol Equity
+    int256 public cachedProtocolEquity;
+
+    /// @notice timestamp of the last read operation on the CollateralizationOracle
+    uint256 public lastUpdate;
+    /// @notice validity duration of the cached value
+    uint256 public validityDuration;
+    /// @notice deviation threshold to consider cached values outdated, in basis
+    ///         points (base 10_000)
+    uint256 public deviationThreshold;
+
+    // ----------- Constructor -----------
+
+    /// @notice CollateralizationOracle constructor
+    /// @param _core Fei Core for reference
+    constructor(
+        address _core
+    ) CoreRef(_core) {}
+
+    // ----------- IOracle override methods -----------
+    /// @notice update reading of the CollateralizationOracle
+    function update() external override whenNotPaused {
```

Comment:
should this method also call update on the CR oracle? I'd lean no but maybe it should

Reply:
let's leave it like this for now